### PR TITLE
linters and format

### LINKED ISSUE #476: Enable linters
Enable golang linter in CI (Github Actions) and other Github apps such as dependabot

### PR DESCRIPTION
**Pull request by fedekunze**
closes #476

### CODE REVIEW

File: golangci.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      linters:
+        - golint
+    - text: "don't use an underscore in package name"
+      linters:
+        - golint
+    - text: "ST1003:"
+      linters:
+        - stylecheck
+    # FIXME: Disabled until golangci-lint updates stylecheck with this fix:
+    # https://github.com/dominikh/go-tools/issues/389
```
Comment (karzak) [inline golangci.yml:53]: Don't really know the context, but is this still relevant? Linked issue is closed and seems resolved.

File: cmd/kvcli/main.go [hunk 2]
```diff
@@ -79,7 +79,7 @@ func main() {
 	}
 }
 
-func queryCmd(cdc *amino.Codec) *cobra.Command {
+func queryCmd(cdc *codec.Codec) *cobra.Command {
```
Comment (karzak) [inline cmd/kvcli/main.go:82]: Why the change here?

Comment (fedekunze) [reply cmd/kvcli/main.go:82]: it's the same type 🤷‍♂️
